model/src/main/kotlin/TextLocation.kt

```diff
@@ -90,4 +92,20 @@ data class TextLocation(
             linesOverlapWith(other) -> 0
             else -> min(abs(other.startLine - endLine), abs(startLine - other.endLine))
         }
+
+    /**
+     * Return a [TextLocation] whose path is relative to [basePath], or throw an [IllegalArgumentException] if the paths
+     * have different roots.
+     */
+    fun withRelativePath(basePath: File): TextLocation {
+        val pathAsFile = File(path)
+
+        val relevantPath = when {
```

Comment:
Should this be called `relativePath` or `relativeFile` instead of `relevantPath`?